.cargo/config.toml

```diff
@@ -1,2 +1,2 @@
 [build]
-rustdocflags = ["--html-in-header", "./doc/katex-header.html"]
+rustdocflags = ["--html-in-header", "./doc/katex-header.html"] # We don't have any `normal` dependency in this package. In most cases, `package.metadata.docs.rs` is recommended.
```

Comment:
(1) What is meant by "normal dependency?"
(2) Do we want to put this config in the `package.metadata.docs.rs` table here, as you mentioned?

Reply:
1. Crates listed in `[dependencies]`. They are called "normal dependencies".

    [`cargo::core::dependency::DepKind::Normal`](https://docs.rs/cargo/0.47.0/cargo/core/dependency/enum.DepKind.html#variant.Normal)

    `cargo_metadata` crate also call them "normal dependencies".

    [`cargo_metadata::DependencyKind::Normal`](https://docs.rs/cargo_metadata/0.11.3/cargo_metadata/enum.DependencyKind.html#variant.Normal) (deserialized from `null`)

    Unlike `cargo rustdoc`, `cargo doc` and attempt to apply `build.rustdocflags` that contain relative file paths for normal dependencies.
    That means, `cd ~/.cargo/registry/src/... && rustdoc --html-in-header ./doc/katex-header.html` is executed.

    ```console
    $ cargo doc -v
    Documenting either v1.6.1
        Running `rustdoc --crate-type lib --crate-name either /home/ryo/.cargo/registry/src/github.com-1ecc6299db9ec823/either-1.6.1/src/lib.rs --cap-lints allow -o /home/ryo/src/github.com/rust-lang-ja/ac-library-rs/target/doc --error-format=json --json=diagnostic-rendered-ansi -L dependency=/home/ryo/src/github.com/rust-lang-ja/ac-library-rs/target/debug/deps --html-in-header ./doc/katex-header.html`
    error: error reading `./doc/katex-header.html`: No such file or directory (os error 2)

    error: Could not document `either`.

    Caused by:
    process didn't exit successfully: `rustdoc --crate-type lib --crate-name either /home/ryo/.cargo/registry/src/github.com-1ecc6299db9ec823/either-1.6.1/src/lib.rs --cap-lints allow -o /home/ryo/src/github.com/rust-lang-ja/ac-library-rs/target/doc --error-format=json --json=diagnostic-rendered-ansi -L dependency=/home/ryo/src/github.com/rust-lang-ja/ac-library-rs/target/debug/deps --html-in-header ./doc/katex-header.html` (exit code: 3)
    ```

2. Maybe we should do that and remove the `build.rustdocflags` configuration. It's rather an unordinary way though `build.rustdocflags` works on docs.rs.

    - <https://docs.rs/rust-latex-doc-minimal-example>
    - <https://github.com/search?q=rustdocflags+language%3ATOML+filename%3Aconfig&type=Code>
    - <https://github.com/search?q=package.metadata.docs.rs+language%3ATOML+filename%3ACargo.toml&type=Code>
    - <https://github.com/rust-lang/docs.rs/blob/0e5e81b8e56cbad604b662141f8c07378542ae40/crates/metadata/lib.rs>